tests/05-sim-long_jumps.py

```diff
@@ -30,21 +31,27 @@
 
 def test(args):
     f = SyscallFilter(KILL)
-    # syscalls referenced by number to make the test simpler
-    f.add_rule_exactly(ALLOW, 1)
+    f.add_rule(ALLOW, "brk")
     i = 0
     while i < 100:
-        f.add_rule_exactly(ALLOW, 1000,
-                           Arg(0, EQ, i),
-                           Arg(1, NE, 0),
-                           Arg(2, LT, sys.maxsize))
+        f.add_rule(ALLOW, "chdir",
+                   Arg(0, EQ, i),
+                   Arg(1, NE, 0),
+                   Arg(2, LT, sys.maxsize))
         i += 1
     i = 100
-    while i < 200:
-        f.add_rule_exactly(ALLOW, i,
-                           Arg(0, NE, 0))
+    ctr = 0
+    while i < 10000 and ctr < 100:
+        if i == resolve_syscall(Arch(), "chdir"):
+            continue
+        try:
+            resolve_syscall(Arch(), i)
+        except ValueError:
+            continue
+        f.add_rule(ALLOW, i, Arg(0, NE, 0))
         i += 1
```

Comment:
I think you could get stuck in an infinite loop here.  If the syscall can't be resolved, we continue above, but don't increment `i`.  You probably need an increment of `i` there as well

Reply:
And, I think you need one in the `chdir` continue spot as well, now that I think of it.

Reply:
Yes, good catch, thanks!